### PR TITLE
Daemonise by default (unless -d is passed on the command line).

### PR DESCRIPTION
**Pull request by ltratt**
This requires a bit of reshuffling of the main method, as something in the `[tokio::main]` setup doesn't work with `daemon` (perhaps it creates threads, as `fork` only carries the main thread over to the child process).

### CODE REVIEW

**Comment by vext01** [comment]
So you didn't have to do anything special to get output to go to syslog?

**Comment by ltratt** [comment]
This PR has nothing to say about syslog, but syslog will be the topic of the next PR after this...

File: src/config.rs [hunk 1]
```diff
@@ -24,6 +24,8 @@ lrlex_mod!("config.l");
 lrpar_mod!("config.y");
 
 pub struct Config {
+    /// If true, then daemonise (i.e. disassociate from the terminal and log errors via syslog).
```
Comment (vext01) [inline src/config.rs]: Because it says here that errors will go to syslog. Perhaps drop that part until the support is implemented?

Comment (ltratt) [reply src/config.rs]: The support is implemented, but I plonked it in a separate PR. I'm inclined to leave this comment as-is -- it's a minor, and very temporary, wart.

Comment (vext01) [reply src/config.rs]: OK

**Pull request merged**